executor/src/lib.rs

```diff
@@ -443,7 +444,7 @@ where
         blocks: Vec<(Vec<Transaction>, Arc<ProcessedVMOutput>)>,
         ledger_info_with_sigs: LedgerInfoWithSignatures,
         synced_trees: &ExecutedTrees,
-    ) -> Result<()> {
+    ) -> Result<Vec<CommittedTransaction>> {
```

Comment:
nit: let's just return original transaction type and do transformation at function call